builder/dockerfile/internals.go

```diff
@@ -248,7 +248,6 @@ func (b *Builder) download(srcURL string) (fi builder.FileInfo, err error) {
 	progressReader := progress.NewProgressReader(resp.Body, progressOutput, resp.ContentLength, "", "Downloading")
 	// Download and dump result to tmp file
 	if _, err = io.Copy(tmpFile, progressReader); err != nil {
-		tmpFile.Close()
```

Comment:
> I don't see why that matters? `tmpFile.Close()` just fails if the file is already closed, so deferring it is just insurance against someone adding more code and forgetting to add the `Close` statements explicitly.

Be careful with this. My understanding is that it's okay to call `Close` twice in the same goroutine, but if different goroutines call it, the calls can race, and one can end up closing an unrelated file. I've hit this problem before.